src/new_index/schema.rs

```diff
@@ -1436,6 +1543,72 @@ impl TxEdgeRow {
     }
 }
 
+struct MWOutputRow {
+    output: mimblewimble::Output
+}
+
+impl MWOutputRow {
+    fn new(output: mimblewimble::Output) -> Self {
+        MWOutputRow { output: output }
+    }
+
+    fn get_key(hash: blake3::Hash) -> Vec<u8> {
+        [b"MWO".to_vec(), hash.as_bytes().to_vec()].concat()
+    }
+
+    fn into_row(self) -> DBRow {
+        DBRow {
+            key: MWOutputRow::get_key(get_output_id(&self.output)),
+            value: bincode::serialize(&self.output).expect("failed to deserialize MWOutputRow"),
+        }
+    }
+
+    fn from_row(row: DBRow) -> Self {
+        MWOutputRow {
+            output: bincode::deserialize(&row.value).unwrap(),
+        }
+    }
+}
+
+struct MWSpentOutputRow {
+    output_id: [u8; 32]
+}
+
+impl MWSpentOutputRow {
+    fn new(output_id: [u8; 32]) -> Self {
+        MWSpentOutputRow { output_id }
+    }
+
+    fn get_key(self) -> Vec<u8> {
+        [b"MWSO".to_vec(), self.output_id.to_vec()].concat()
+    }
+
+    fn into_row(self) -> DBRow {
+        DBRow {
+            key: self.get_key(),
+            value: vec! [],
+        }
+    }
+
+    fn from_row(row: DBRow) -> Self {
+        MWSpentOutputRow {
+            output_id: std::convert::TryInto::try_into(row.key).unwrap(),
+        }
+    }
+}
+
+fn get_output_id(output: &mimblewimble::Output) -> blake3::Hash {
```

Comment:
nit: You'll probably need this lots of other places as well. Consider moving it into mimblewimble::Output, and if the structure is immutable, I'd recommend caching the result.

Reply:
Right now it's used only in one place.